### PR TITLE
Problem: missing grpc integration test (fix #194)

### LINKED ISSUE #194: Problem: missing grpc integration test
add grpc integrataion test
1. make tx
2. sign tx
3. broadcast with grpc

### PR DESCRIPTION
**Pull request by leejw51crypto**
Solution: add grpc query, tx broadcast integration test
generate proto+grpc code on the fly

### CODE REVIEW

**Comment by leejw51crypto** [comment]
fixing 
- nix file lint issue
- grpc_python_plugin absolute path issue

File: pystarport/pystarport/convert.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+OUTPUT=./proto_python
+COSMOS=./cosmos-sdk
+TENDERMINT=./tendermint
+TMP=$(whereis grpc_python_plugin)
+PLUGIN="$(cut -d' ' -f2 <<<"$TMP")"
+mkdir $OUTPUT
+git clone https://github.com/cosmos/cosmos-sdk.git
+git clone https://github.com/tendermint/tendermint.git
```
Comment (tomtau) [inline pystarport/pystarport/convert.sh:9]: not sure whether to pin this against master (which can be non-reproducible, but can possibly find out issues earlier) or against specific commits/tags (e.g. v0.40-rc2 for SDK) to match our code

Comment (tomtau) [reply pystarport/pystarport/convert.sh:9]: `git checkout <tag>`

**Comment by leejw51crypto** [comment]
fixed lint issue, and integration test

**Review by tomtau — COMMENTED** [review]
looks ok -- just one comment whether to pin the makeproto script against specific versions

File: pystarport/pystarport/convert.sh (hunk 1 above)
Comment (leejw51crypto) [reply pystarport/pystarport/convert.sh:9]: ok, i'll pin the version

File: pystarport/pystarport/default.nix [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{ pkgs ? import ../../nix { }, commit ? "" }:
```
Comment (yihuang) [inline pystarport/pystarport/default.nix:1]: - this is the directory of python package, better to put config files outside
- maybe we make the proto generated files as a nix package, which can be depend by pystarport package?

File: .flake8 [hunk 3]
```diff
@@ -1,3 +1,4 @@
 [flake8]
 max-line-length = 88
 extend-ignore = E203
+exclude = .git,__pycache__,./pystarport/pystarport
```
Comment (yihuang) [inline .flake8:4]: you excluded the whole pystarport package from python linter?

File: pystarport/pystarport/convert.sh (hunk 1 above)
Comment (tomtau) [reply pystarport/pystarport/convert.sh:9]: or `git clone -b v0.40-rc2  https://github.com/cosmos/cosmos-sdk.git` && `git clone -b v0.34-rc5  https://github.com/tendermint/tendermint.git`

File: .flake8 (hunk 3 above)
Comment (leejw51crypto) [reply .flake8:4]: i'll change it, exclude only cosmos, tendermint

File: pystarport/pystarport/default.nix (hunk 2 above)
Comment (leejw51crypto) [reply pystarport/pystarport/default.nix:1]: i'll try not to use ../../nix

File: pystarport/pystarport/convert.sh (hunk 1 above)
Comment (leejw51crypto) [reply pystarport/pystarport/convert.sh:9]: ok, looking good

Comment (yihuang) [reply pystarport/pystarport/convert.sh]: or `curl ...archive/hash.tar.gz`

**Review by tomtau — APPROVED** [review]

**Pull request merged**